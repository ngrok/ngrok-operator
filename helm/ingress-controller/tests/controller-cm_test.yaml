suite: test controller-cm
templates:
- controller-cm.yaml
release:
  name: test-release
  namespace: test-namespace
tests:
- it: should match snapshot
  asserts:
  - matchSnapshot: {}
- it: should create a configmap
  asserts:
  - isKind:
      of: ConfigMap
  - isAPIVersion:
      of: v1
  - hasDocuments:
      count: 1
- it: should contain the correct namespace
  set:
    namespace: test-namespace
  asserts:
  - equal:
      path: metadata.namespace
      value: test-namespace
- it: Should create a configmap with the correct name
  asserts:
  - equal:
      path: metadata.name
      value: test-release-kubernetes-ingress-controller-manager-config
- it: Works when fullname is supplied
  set:
    fullnameOverride: fno
  asserts:
  - equal:
      path: metadata.name
      value: fno-manager-config
