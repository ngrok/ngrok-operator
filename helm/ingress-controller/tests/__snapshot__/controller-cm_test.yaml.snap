should match snapshot:
  1: |
    apiVersion: v1
    data:
      controller_manager_config.yaml: |
        apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
        kind: ControllerManagerConfig
        health:
          healthProbeBindAddress: :8081
        metrics:
          bindAddress: 127.0.0.1:8080
        leaderElection:
          leaderElect: true
          resourceName: test-release-kubernetes-ingress-controller-leader
    kind: ConfigMap
    metadata:
      name: test-release-kubernetes-ingress-controller-manager-config
      namespace: test-namespace
